addons/t9n/views/t9n_resource_views.xml

```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0"?>
+<odoo>
+    <data>
+        <record id="t9n.resource_view_form" model="ir.ui.view">
+            <field name="name">Resource</field>
+            <field name="model">t9n.resource</field>
+            <field name="arch" type="xml">
+                <form string="Resource Form">
+                    <sheet>
+                        <div>
+                            <field name="file_name" invisible="True"/>
+                            <field name="file" filename="file_name" widget="field_binary_file"/>
+                            <field name="message_ids"/>
+                        </div>
+                    </sheet>
+                </form>
+            </field>
+        </record>
+    </data>
+    <record id="t9n.resource_view_tree" model="ir.ui.view">
+        <field name="name">t9n.resource.tree</field>
+        <field name="model">t9n.resource</field>
+        <field name="arch" type="xml">
+            <tree>
+                <field name="file" filename="file_name" widget="field_binary_file" string="Resource File"/>
+                <field name="file_name" invisible="True" string=""/>
```

Comment:
why is it invisible?

Reply:
Because if it's visible in the form view it will give you the ability to set the file name yourself. We need the file_name field to be not editable. It can be achieved also through `readonly` attribute.

Reply:
It should be visible with a `readonly` attribute just to show the file name but no editing it.